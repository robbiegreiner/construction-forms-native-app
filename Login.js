import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  TextInput
} from 'react-native';


export default class Login extends Component<{}> {
  constructor(){
    super()
    this.state = {
      email: '',
      password: ''
    };
  }


  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.login}>
          Login Here
        </Text>
        <Text>
          {this.state.email}
        </Text>
        <Text>
          {this.state.password}
        </Text>
        <TextInput
          style={{height:40}}
          placeholder="email"
          onChangeText={(text) => this.setState({email: text})}
        />
        <TextInput
          style={{height:40}}
          placeholder="password"
          onChangeText={(text) => this.setState({password: text})}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {

  },
  login: {
    textAlign: 'center',
    color: '#333333',
    fontSize: 24,
    marginBottom: 5,
  },
});
