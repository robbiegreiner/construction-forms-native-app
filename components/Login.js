import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  TextInput,
  Button,
  Alert
} from 'react-native';

import firebase from '../firebase';


export default class Login extends Component<{}> {
  constructor(){
    super()
    this.state = {
      email: '',
      password: ''
    };
  }

  login(email, password) {
    firebase.auth().signInWithEmailAndPassword(email,password)
      .then(response => this.props.setUser(response.email))
      //catch error here
  }


  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.login}>
          Login Here
        </Text>
        <TextInput
          style={{height:40}}
          placeholder="email"
          onChangeText={(text) => this.setState({email: text})}
        />
        <TextInput
          style={{height:40}}
          placeholder="password"
          onChangeText={(text) => this.setState({password: text})}
        />
        <Button
          onPress={() => this.login(this.state.email, this.state.password)}
          title="Login"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {

  },
  login: {
    textAlign: 'center',
    color: '#333333',
    fontSize: 24,
    marginBottom: 5,
  },
});
