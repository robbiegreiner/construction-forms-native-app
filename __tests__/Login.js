/* eslint-disable */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../components/Login';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, configure } from 'enzyme';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer.create(<Login />);
  expect(tree).toBeDefined();
});

it('maches snapshot', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should have a default state', () => {
  const wrapper = shallow(<Login />)
  expect(wrapper.state()).toEqual({
    email: '',
    password: '',
    createAccount: null,
  });
});
