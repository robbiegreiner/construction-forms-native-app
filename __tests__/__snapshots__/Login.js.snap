// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E8C712",
      "justifyContent": "center",
      "marginTop": 100,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 36,
        "textAlign": "center",
      }
    }
  >
    Construction Forms
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      Object {
        "height": 150,
        "marginBottom": 20,
        "marginTop": 20,
        "width": 150,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "fontSize": 24,
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Login Here
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="email"
    style={
      Object {
        "height": 40,
        "width": 200,
      }
    }
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="password"
    secureTextEntry={true}
    style={
      Object {
        "height": 40,
        "width": 200,
      }
    }
    value=""
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Create Account Here
      </Text>
    </View>
  </View>
</View>
`;
